### ISSUE TITLE
What is the correct require() pattern for the plugins ?

### ISSUE THREAD

**Issue by lunelson** [aFarkas/lazysizes#267]
I notice there's a `-umd` version of the script; however I can't find any reference to the **exact pattern for bundling that script with plugins**, using `require()`.

The following typical `<script>` order also works as an order for concatenation;

``` html
<script src="../plugins/parent-fit/ls.parent-fit.min.js"></script>
<script src="../plugins/object-fit/ls.object-fit.min.js"></script>
<script src="../plugins/respimg/ls.respimg.min.js"></script>
<script src="../lazysizes.min.js" async=""></script>
```

but what is it right for `require()`? Do I need to make a global assignment, and if so, how to bring in the plugins correctly?

``` js
// e.g. browserify
global.lazysizes = require("../lazysizes-umd.min.js");
require("../plugins/parent-fit/ls.parent-fit.min.js");
require("../plugins/object-fit/ls.object-fit.min.js");
require("../plugins/respimg/ls.respimg.min.js");
```

**Comment by aFarkas** [comment]
You can do this how you want. You also don't need a global assignment.

**Comment by lunelson** [comment]
So you mean, the pattern I listed above, without the global assignment, is correct? i.e. that the `lazySizes` global variable is created automatically, and the plugins are added correctly, if I do as follows? If so, I will make a PR on the documentation to indicate this.
```js
// e.g. browserify
require("../lazysizes-umd.min.js");
require("../plugins/parent-fit/ls.parent-fit.min.js");
require("../plugins/object-fit/ls.object-fit.min.js");
require("../plugins/respimg/ls.respimg.min.js");
```

**Comment by aFarkas** [comment]
Yes, but if you use browserify or webpack you can reference the module itself.

```js
require('lazysizes');
require('lazysizes/plugins/optimumx/ls.optimumx');
require('lazysizes/plugins/parent-fit/ls.parent-fit');
```

**Comment by rdunk** [comment]
I'm trying to use the unveilhooks plugin with webpack using the following.

```
require('lazysizes');
require('lazysizes/plugins/unveilhooks/ls.unveilhooks');
```

But keep getting this error:

> WARNING in ./~/lazysizes/plugins/unveilhooks/ls.unveilhooks.js
> Critical dependencies:
> 73:6-20 the request of a dependency is an expression
>  @ ./~/lazysizes/plugins/unveilhooks/ls.unveilhooks.js 73:6-20

Am I missing something?

**Comment by AurelioDeRosa** [comment]
I get this error as well with `ls.unveilhooks.js`

**Comment by hilja** [comment]
I also get the warning from webpack when including `ls.unveilhooks.js`:

```
WARNING in ./~/lazysizes/plugins/unveilhooks/ls.unveilhooks.js
Critical dependencies:
73:6-20 the request of a dependency is an expression
@ ./~/lazysizes/plugins/unveilhooks/ls.unveilhooks.js 73:6-20
```

It goes away if I remove [this require call](https://github.com/aFarkas/lazysizes/blob/gh-pages/plugins/unveilhooks/ls.unveilhooks.js#L74):

```
// handle data-require
tmp = e.target.getAttribute('data-require');
if(tmp){
    if(window.require){
       // This causes the warning 
       require([tmp]);
    }
}
```

**Comment by aFarkas** [comment]
@hilja 
Thanks for the information.

Do you guys know wether and how it is possible to use requirejs in a browserify/webpack enviroment?

**Comment by synapticism** [comment]
I just struck upon the same issue with both unveilhooks/include. The problem is that Webpack requires statically analyzable `require` calls to be able to build bundles in advance. This is [mentioned in the docs](https://webpack.github.io/docs/webpack-for-browserify-users.html#external-requires) (along with a workaround that appears too clunky at first glance for me to want to mess with). Not sure if a fix is feasible after toying around with it for about half an hour.

**Comment by aFarkas** [comment]
I think I will remove the require calls in both scripts. And add a default loadScript hook, that can be configured by developers to use `require` if needed.

**Comment by aFarkas** [comment]
I just removed the inline `require` calls. And released a new version `"3.0.0-rc1"`. Please test if you have time.
